File: 1task.py
def cyclic_shift(s: str, t: str) -> bool:
    if len(s) != len(t):
        return False
    doubled = s + s
    return t in doubled

def main():
    s = input("S: ")
    t = input("T: ")

    if cyclic_shift(s, t):
        print("Yes")
    else:
        print("No")

File: 1task.go
package main

import (
	"fmt"
	"strings"
)

func CyclicShift(S, T string) bool {
	if len(S) != len(T) {
		return false
	}
	doubled := S + S
	return strings.Contains(doubled, T)
}

func main() {
	var S, T string

	fmt.Println("S:")
	fmt.Scan(&S)

	fmt.Println("T:")
	fmt.Scan(&T)

	if CyclicShift(S, T) {
		fmt.Println("Yes")
	} else {
		fmt.Println("No")
	}
}